django/db/backends/base/features.py

```diff
@@ -281,6 +281,10 @@ class BaseDatabaseFeatures:
     supports_functions_in_partial_indexes = True
     # Does the backend support covering indexes (CREATE INDEX ... INCLUDE ...)?
     supports_covering_indexes = False
+    # Does the backend support indexes on expressions?
+    supports_expression_indexes = False
```

Comment:
I'd suggest defaulting to True so databases have to opt-out rather than opt-in. It would be good to mention this in "Database backend API" under "Backwards incompatible changes in 3.2"  like other new features are.

Reply:
I agree that it should be documented, it was probably removed a little to fast by me when we changed the default of `supports_expression_indexes ` from `True` to `False`. Not sure about the changing default back to `True`, my gut feeling is that it'll cause more trouble by being opt-out for less maintained backends, but you have far more experience with developing third-party-backends 😄 

It'll open up a PR which starts with adding the documentation and we can continue the discussion about the default there.

Reply:
https://github.com/django/django/pull/13885